godtools/App/Services/Renderer/ManifestResourcesCache/ManifestResourcesCache.swift

```diff
@@ -39,9 +40,13 @@ class ManifestResourcesCache: ManifestResourcesCacheType {
         return translationsFileCache.getFile(location: location)
     }
     
-    func getImageFromManifestResources(fileName: String) -> UIImage? {
+    func getImageFromManifestResources(resource: Resource) -> UIImage? {
         
-        guard let location = getSHA256FileLocation(fileName: fileName) else {
+        guard let resourceName = resource.name, !resourceName.isEmpty else {
+            return nil
+        }
+        
+        guard let location = getSHA256FileLocation(fileName: resourceName) else {
```

Comment:
it looks like you will have to move the current fileName -> resource lookup logic into `getFile()` since that also calls `getSHA256FileLocation()`. That code path might be able to be improved to pass in a `resource` object as well?